File: Interface/python/src/app/api/envelope.py
"""세 인터페이스가 공유하는 응답 envelope {success, data, error}."""

from __future__ import annotations

from typing import Any

from fastapi.responses import JSONResponse


def success(data: Any, status_code: int = 200) -> JSONResponse:
    return JSONResponse(status_code=status_code, content={"success": True, "data": data})


def failure(code: str, message: str, status_code: int) -> JSONResponse:
    return JSONResponse(
        status_code=status_code,
        content={"success": False, "error": {"code": code, "message": message}},
    )

File: Interface/js/src/common/envelope.ts
// 세 인터페이스가 공유하는 응답 envelope {success, data, error}
export interface ErrorBody {
  code: string;
  message: string;
}

export interface Envelope<T> {
  success: boolean;
  data?: T;
  error?: ErrorBody;
}

export function successEnvelope<T>(data: T): Envelope<T> {
  return { success: true, data };
}

export function errorEnvelope(code: string, message: string): Envelope<never> {
  return { success: false, error: { code, message } };
}
